src/operators.jl

```diff
@@ -115,9 +115,19 @@ function union{T<:AbstractGraph}(g::T, h::T)
     return r
 end
 
+isunique(iter::Range) = true
+isunique(iter::Set) = true
+isunique(iter) = (length(unique(iter)) == length(iter))
+
 #@doc "filter g to include only the vertices present in iter which should not have duplicates
 #returns the subgraph of g induced by set(iter) along with the mapping from the old vertex names to the new vertex names" ->
-function inducedsubgraph{T<:AbstractGraph}(g::T, iter)
+function induced_subgraph{T<:AbstractGraph}(g::T, iter)
+    !isunique(iter) && error("Vertices in subgraph list must be unique")
```

Comment:
I didn't think about the issue with ordering but unique preserves order by https://github.com/JuliaLang/julia/commit/68e0398bf29aece96b059a5b758f13031192deac

It is pretty clear that the right thing to do is to use `unique(iter)`. In order to use the version that throws an error safely you have to either try/catch or do `g[unique(iter))`.
Can you see someone catching that error and doing something else? Or are we just helping people catch bugs in the calling code?